packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -114,6 +102,22 @@ const experimentalStyled = (tag, options, muiOptions = {}) => {
       });
     }
 
+    if (Array.isArray(styleArg)) {
```

Comment:
We could even add a condition before this, soemthing like:

```
if (expressionsWithDefaultTheme.length - expressions.length) { // if we added custom style functions
```

Also please let's store this in some constant, like `numOfCustomFnsApplied` or something shorter :)